### PR TITLE
Fixes/error handling

### PR DESCRIPTION
**Pull request by vivekimsit**
Changelog:

* Use styled component for styling.

* Use Redis for session storage

* Better error handling

### CODE REVIEW

File: frontend/src/API/api.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import axios from 'axios';
```
Comment (Mortuie) [inline frontend/src/API/api.js:1]: Are we using axios or fetch @nikrb i prefer axios but up to debate

File: frontend/src/API/user.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import api from './api'
+
+export const KEY = 'HISSUES';
```
Comment (Mortuie) [inline frontend/src/API/user.js]: This is repeated and makes having a constant useless

File: frontend/src/API/service.js [hunk 3]
```diff
@@ -1,4 +1,3 @@
-
 const API_ROOT = '/api/v1';
```
Comment (Mortuie) [inline frontend/src/API/service.js:1]: we have an api_root and a base URL in our axios config? which one get used...?

File: frontend/src/Dashboard/Dashboard.Loader.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ import {StyleSheet, css} from 'aphrodite';
 const DashboardLoader = () => (
   <div style={{flex: '1'}}>
     <h1>Dashboard</h1>
-    <button type="button" disabled="true" className={css(styles.button)}/>
+    <button type="button" disabled={true} className={css(styles.button)}/>
```
Comment (Mortuie) [inline frontend/src/Dashboard/Dashboard.Loader.js]: This is still aphrodite isn't it

File: frontend/src/Dashboard/Dashboard.js [hunk 5]
```diff
@@ -1,33 +1,49 @@
...
+
+import { Footer } from '../Footer';
+import { houseIssuesFetchData, houseDelete } from './actions';
+import { HouseList } from '../House';
 import DashboardLoader from './Dashboard.Loader';
 
 class Dashboard extends Component {
   static propTypes = {
+    currentUser: PropTypes.object.isRequired,
     houseIssues: PropTypes.arrayOf( PropTypes.object),
```
Comment (Mortuie) [inline frontend/src/Dashboard/Dashboard.js]: remove space please.

File: frontend/src/Dashboard/Dashboard.js [hunk 6]
```diff
@@ -1,33 +1,49 @@
...
     deleteHouse: PropTypes.func.isRequired
   };
+
   state = {
     redirect_new_house: false
-  };
+  }
+
   componentDidMount = () => {
     this.props.fetchData( '/api/v1/house-issues');
```
Comment (Mortuie) [inline frontend/src/Dashboard/Dashboard.js]: Random spaces throughout this file...

File: frontend/src/Dashboard/Dashboard.js [hunk 7]
```diff
@@ -1,33 +1,49 @@
...
+
+  render() {
+    const { currentUser } = this.props;
+    const isLoggedIn = !isEmpty(currentUser);
+
+    if(!isLoggedIn) {
+      console.error('Invalid state, should not be here');
+    }
+
+    if(this.state.redirect_new_house) {
```
Comment (Mortuie) [inline frontend/src/Dashboard/Dashboard.js]: space `if (this.state.redirect)`

File: frontend/src/Dashboard/Dashboard.js [hunk 8]
```diff
@@ -47,24 +63,22 @@ class Dashboard extends Component {
         <h1>Dashboard</h1>
         <button type="button" onClick={this.onNewHouse} >New House</button>
```
Comment (Mortuie) [inline frontend/src/Dashboard/Dashboard.js]: random space at the end of onClick.

File: frontend/src/Dashboard/Dashboard.js [hunk 9]
```diff
@@ -47,24 +63,22 @@ class Dashboard extends Component {
...
 
-const mapDispatchToProps = dispatch => {
-  return {
-    deleteHouse: (house_id) => dispatch( houseDelete(house_id)),
-    fetchData: (url) => dispatch( houseIssuesFetchData(url))
-  };
-};
+const mapDispatchToProps = dispatch => ({
+  deleteHouse: (house_id) => dispatch( houseDelete(house_id)),
+  fetchData: (url) => dispatch( houseIssuesFetchData(url))
```
Comment (Mortuie) [inline frontend/src/Dashboard/Dashboard.js]: random space

File: frontend/src/Footer/Footer.js [hunk 10]
```diff
@@ -1,23 +1,20 @@
+import styled from 'styled-components';
 import React from 'react';
-import { css, StyleSheet } from 'aphrodite';
 
-const Footer = () => (
-  <footer className={css(styles.footer)}>
-    <section className={css(styles.section)}>
+export const Footer = styled.footer`
```
Comment (Mortuie) [inline frontend/src/Footer/Footer.js]: Why are you exporting the footer style???

File: frontend/src/Navbar/index.js [hunk 11]
```diff
@@ -1,3 +1,98 @@
...
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import { StyleSheet, css } from 'aphrodite';
+import { connect } from 'react-redux';
+import isEmpty from 'lodash/isEmpty';
 
-export { Navbar };
\ No newline at end of file
+
+class C extends Component {
```
Comment (Mortuie) [inline frontend/src/Navbar/index.js]: Class C ?? Please rename

**Review by Mortuie — DISMISSED** [review]

File: frontend/src/API/api.js (hunk 1 above)
Comment (vivekimsit) [reply frontend/src/API/api.js:1]: I prefer axios.

Comment (Mortuie) [reply frontend/src/API/api.js:1]: @nikrb

File: frontend/src/API/service.js (hunk 3 above)
Comment (vivekimsit) [reply frontend/src/API/service.js:1]: I am not sure about the old code but I will prefer to use later one everywhere.

File: frontend/src/API/api.js (hunk 1 above)
Comment (nikrb) [reply frontend/src/API/api.js:1]: oh sorry, I don't mind, axios is good @Mortuie

**Comment by vivekimsit** [comment]
It also fixes following issues as well:

fixes #114 
fixes #116

**Review by nikrb — APPROVED** [review]
let's do it

**Pull request merged**